gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py

```diff
@@ -154,3 +222,95 @@ def _open_ds(
         open_options = ["{}={}".format(k, v) for k, v in open_options.items()]
 
         return gdal.OpenEx(str(filename), access_mode, open_options=open_options)
+
+
+def get_data_type(data_type: Optional[DataTypeOrStr]):
+    if data_type is None:
+        return None
+    if isinstance(data_type, str):
+        return gdal.GetDataTypeByName(data_type)
+    else:
+        return data_type
+
+
+def get_raster_bands(ds: gdal.Dataset) -> Iterator[gdal.Band]:
+    return (ds.GetRasterBand(i + 1) for i in range(ds.RasterCount))
+
+
+def get_band_types(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds) as ds:
+        return [band.DataType for band in get_raster_bands(ds)]
+
+
+def get_band_minimum(band: gdal.Band):
+    ret = band.GetMinimum()
+    if ret is None:
+        band.ComputeStatistics(0)
+    return band.GetMinimum()
+
+
+def get_raster_band(filename_or_ds: PathOrDS, bnd_index: int = 1, ovr_index: Optional[int] = None):
+    with OpenDS(filename_or_ds) as ds:
+        bnd = ds.GetRasterBand(bnd_index)
+        if ovr_index is not None:
+            bnd = bnd.GetOverview(ovr_index)
+        return bnd
+
+
+def get_raster_minimum(filename_or_ds: PathOrDS, bnd_index: Optional[int] = 1):
+    with OpenDS(filename_or_ds) as ds:
+        if bnd_index is None:
+            return min(get_band_minimum(bnd) for bnd in get_raster_bands(ds))
+        else:
+            bnd = ds.GetRasterBand(bnd_index)
+            return get_band_minimum(bnd)
+
+
+def get_raster_min_max(filename_or_ds: PathOrDS, bnd_index: int = 1):
+    with OpenDS(filename_or_ds) as ds:
+        bnd = ds.GetRasterBand(bnd_index)
+        bnd.ComputeStatistics(0)
+        min_val = bnd.GetMinimum()
+        max_val = bnd.GetMaximum()
+        return min_val, max_val
+
+
+def get_nodatavalue(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds) as ds:
+        band = next(get_raster_bands(ds))
+        return band.GetNoDataValue()
+
+
+def unset_nodatavalue(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds, access_mode=gdal.GA_Update) as ds:
+        for b in get_raster_bands(ds):
+            b.DeleteNoDataValue()
+
+
+def get_image_structure_metadata(filename_or_ds: PathOrDS, key: str, default=None):
```

Comment:
could be simplified as ds.GetMetadataItem(key, "IMAGE_STRUCTURE")

Reply:
thanks